crates/cow-amm/src/cache.rs

```diff
@@ -26,6 +33,47 @@ impl Storage {
             .flat_map(|amms| amms.iter().cloned())
             .collect()
     }
+
+    pub(crate) async fn drop_empty_amms(&self, web3: &Web3) {
```

Comment:
I don't think storage should depend on Web3 (it's just a storage component). It already exposes all amms, which can be queried in the new maintenance component you created. Storage should then only expose a remove method which can be called from that component.